### DISCUSSION TITLE
Private package used in eslint config

### DISCUSSION THREAD

**Discussion by bryanjtc** [oxsecurity/megalinter#2021] [Q&A] [answered]
My eslint config uses a private package. I use .yarnc.yml to set the npmAuthToken needed to fetch the package. The megalinter throws an error in the eslint config because it can't find the package. How do I fix it? Do I need a .npmrc file and a custom command to run before the linter?

**Comment by nvuillam** [comment, marked as answer]
I've never played with npm private packages, but I suppose you can manage its installation using PRE_COMMANDS

https://oxsecurity.github.io/megalinter/latest/configuration/#pre-commands

**Reply by bryanjtc** [reply]
I will try.

**Reply by bryanjtc** [reply]
@nvuillam Can I use the environment variables from a .env file or variables set in the ci/cd pipeline on the mega-linter.yml?

**Reply by Kurt-von-Laven** [reply]
I think you would have to load the `.env` file yourself during `PRE_COMMANDS`.

**Reply by nvuillam** [reply]
If you define SOME_VARIABLE in CI variables, it should be available in a pre_command `echo "$SOME_VARIABLE"`
